### PR TITLE
Bump required Maven version to 3.2

### PR DESCRIPTION
**Pull request by swegner**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -90,10 +90,6 @@
     </developer>
   </developers>
 
-  <prerequisites>
```
Comment (swegner) [inline pom.xml]: Prerequisites are not actually enforced in Maven 3, see: http://stackoverflow.com/a/18058572/33791

File: pom.xml [hunk 2]
```diff
@@ -1017,7 +1013,7 @@
                   <version>[1.7,)</version>
                 </requireJavaVersion>
                 <requireMavenVersion>
-                  <version>[3.0.3,)</version>
```
Comment (swegner) [inline pom.xml]: `maven-os-plugin`, used in gcp-io, requires at least Maven 3.1. As long as we're bumping the dependency, I propose we go to 3.3, which was [released March 2015](https://maven.apache.org/docs/history.html).

**Comment by swegner** [comment]
R: @kennknowles

File: pom.xml (hunk 2 above)
Comment (swegner) [inline pom.xml]: Looks like Travis provides [Maven 3.2](https://docs.travis-ci.com/user/ci-environment/#Maven-version). If Travis isn't yet on 3.3, perhaps we can't expect it of our users either. 

Lowering the requirement to 3.2, which was released 2014-02-21.

**Comment by kennknowles** [comment]
I now remember what feature I really wanted, and it is available as of 3.2.1: Being able to explicitly exclude build of a project in the Reactor, since dependency exclusions don't work there.

**Comment by kennknowles** [comment]
Another reference point is Ubuntu release history (other flavors also matter, of course). https://launchpad.net/ubuntu/+source/maven. Looks like a jump from 3.0 to 3.3 on 2015-08-05. So anyone using an OS that is still < 3.2 is probably used to having to install their own software, and it won't be too great a burden.

LGTM. Certainly we shouldn't have the incorrect version there. If there are any complaints, we can lower to 3.1.

**Pull request merged**